server/routes/api/urls/urls.ts

```diff
@@ -75,12 +76,24 @@ router.post(
     }
 
     // External resources
+    const cachedData = await CacheHelper.getData(CacheHelper.getUnfurlKey(url));
+    if (cachedData) {
+      return (ctx.body = presentUnfurl(cachedData));
+    }
+
     for (const plugin of plugins) {
-      const data = await plugin.value(url);
+      const data = await plugin.value.unfurl(url);
       if (data) {
-        return "error" in data
-          ? (ctx.response.status = 204)
-          : (ctx.body = presentUnfurl(data));
+        if ("error" in data) {
+          return (ctx.response.status = 204);
+        } else {
+          await CacheHelper.setData(
+            CacheHelper.getUnfurlKey(url),
```

Comment:
As the cache now includes authenticated integrations the cache key must include the team ID to stop the cache being shared between workspaces.

Another advantage of this would be that we can clear the cache related to a specific team when they enable new unfurl integrations.